### PR TITLE
Add a global action to add a new file in the explorer - resolves #5547

### LINKED ISSUE #5547: Add a global action to add a new file in the explorer
- VSCode Version: 1.0.0
- OS Version: OSX 10.11.3

I noticed that we can set `workbench.action.files.newFolder` to a keybinding. Which creates a nice input field in the sidebar to create the directory at the current selected position.

screen shot 2016-04-20 at 15 06 05

But there is no possibility to do this for a file. Yes you can bind `workbench.action.files.newUntitledFile` but it does not do the same thing. It just creates a new `untitled file` and when pressing <kbd>cmd</kbd> + <kbd>s</kbd> you have to choose the destination in a picker. 

So, is it possible to add something for a new file as well? It would really improve my speed because I don't have to touch my mouse anymore to create a new named file in a specific directory.

### PR DESCRIPTION
**Pull request by SamVerschueren**
Adds the possibility of binding a keyboard shortcut to a `Create new file` action. Resolves #5547 

@bpasero I believe this should do the trick :)

### CODE REVIEW

File: src/vs/workbench/parts/files/electron-browser/files.electron.contribution.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ const category = nls.localize('filesCategory', "Files");
 
 let workbenchActionsRegistry = <IWorkbenchActionRegistry>Registry.as(ActionExtensions.WorkbenchActions);
 workbenchActionsRegistry.registerWorkbenchAction(new SyncActionDescriptor(SaveFileAsAction, SaveFileAsAction.ID, SaveFileAsAction.LABEL, { primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KEY_S }), category, ['save', 'as']);
-workbenchActionsRegistry.registerWorkbenchAction(new SyncActionDescriptor(GlobalNewFileAction, GlobalNewFileAction.ID, GlobalNewFileAction.LABEL, { primary: KeyMod.CtrlCmd | KeyCode.KEY_N }), category, ['new', 'file']);
+workbenchActionsRegistry.registerWorkbenchAction(new SyncActionDescriptor(GlobalNewUntitledFileAction, GlobalNewUntitledFileAction.ID, GlobalNewUntitledFileAction.LABEL, { primary: KeyMod.CtrlCmd | KeyCode.KEY_N }), category, ['new', 'file']);
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/files.electron.contribution.ts]: Maybe add the "untitled" to the keywords?

File: src/vs/workbench/parts/files/browser/fileActions.ts [hunk 2]
```diff
@@ -560,7 +560,7 @@ export abstract class BaseGlobalNewAction extends Action {
 }
 
 /* Create new file from anywhere: Open untitled */
-export class GlobalNewFileAction extends Action {
+export class GlobalNewUntitledFileAction extends Action {
 	public static ID = 'workbench.action.files.newUntitledFile';
 	public static LABEL = nls.localize('newFile', "New File");
```
Comment (bpasero) [inline src/vs/workbench/parts/files/browser/fileActions.ts]: Should this maybe read "New Untitled File" to make the difference clear?

**Comment by bpasero** [comment]
@SamVerschueren cool, added some comments!

File: src/vs/workbench/parts/files/browser/fileActions.ts (hunk 2 above)
Comment (SamVerschueren) [inline src/vs/workbench/parts/files/browser/fileActions.ts]: That would definitely be better! How can I do that as probably it should become `('newUntitledFile', "New Untitled File")` and those translations should be adjusted accordingly right? Or are the translations done separately before release?

File: src/vs/workbench/parts/files/electron-browser/files.electron.contribution.ts (hunk 1 above)
Comment (SamVerschueren) [inline src/vs/workbench/parts/files/electron-browser/files.electron.contribution.ts]: 👍

**Comment by SamVerschueren** [comment]
Thanks for the feedback, I adjusted accordingly. The only concern is the [translation](https://github.com/Microsoft/vscode/pull/5597#discussion_r60563979) of `newUntitledFile`.

**Comment by bpasero** [comment]
👍

**Comment by SamVerschueren** [comment]
Awesome, thanks @bpasero for helping me out and merging my PR :).

**Comment by bpasero** [comment]
@SamVerschueren expecting a PR from every issue you file from now on :)

**Comment by SamVerschueren** [comment]
Hehe, I'll do my best :). The project is quite complex to be honest :p.

**Comment by bpasero** [comment]
Welcome to our life ;)

**Pull request merged**